schema/Readme.md

```diff
@@ -920,73 +932,195 @@ model User {
+type Boolean @raw("boolean") @transport("boolean")
+type Bool = Boolean
+type Point @raw("point") @transport("int128")
+type Line @raw("line") @transport("int256")
+type LSeg @raw("lseg") @transport("int256")
+type Box @raw("box") @transport("int256")
+type Path @raw("path") @transport("[]int")
+type Polygon @raw("polygon") @transport("[]int")
+type CIDR @raw("cidr") @transport("[]int")
+type INet @raw("inet") @transport("[]int")
+type Macaddr @raw("macaddr") @transport("int48")
+type Bit(n Int) @raw("bit($n)") @transport("int($n)")
+type BitVarying @raw("bit varying($n)") @transport("[]int")
+type VarBit = BitVarying
+type TSVector @raw("tsvector") @transport("[]int")
+type TSQuery @raw("tsquery") @transport("[]int")
+type UUID @raw("uuid") @transport("int128")
+type XML @raw("xml") @transport("[]int")
+type JSON @raw("json") @transport("[]int")
+type JSONB @raw("jsonb") @transport("[]int")
+type Int4Range @raw("int4range") @transport("int32")
+type Int8Range @raw("int8range") @transport("int64")
+type NumRange @raw("numrange") @transport("[]int")
+type TSRange @raw("tsrange") @transport("int128")
+type TSTZRange @raw("tstzrange") @transport("int96")
+type DateRange @raw("daterange") @transport("int32")
+type PGLSN @raw("pg_lsn") @transport("int64")
+type TXIDSnapshot @raw("txid_snapshot") @transport("[]byte")
+```
+
+#### mysql.prisma (generated)
+
+```groovy
+// standard types
+type String = Text
+type Boolean = Boolean
+type Int = Int
+type Float = Float
+type Timestamp = Timestamp
+
+// root type definitions
+type TinyInt @raw("tinyint") @transport("int1")
+type SmallInt @raw("smallint") @transport("int2")
+type MediumInt @raw("mediumint") @transport("int3")
+type Int @raw("int") @transport("int4")
+type BigInt @raw("bigint") @transport("int8")
+type Float @raw("float") @transport("float32")
+type Double @raw("double") @transport("float64")
+type Timestamp @raw("timestamp") @transport("") // TODO transport
+type Text @raw("text") @transport("[]byte")
+type Boolean @raw("boolean") @transport("boolean")
+type VarChar(length Int) @raw("varchar($length)") @transport("[]byte")
+```
+
+#### schema.prisma
 
 ```groovy
 datasource pg {
-  provider = "postgres"
-  url = "postgres://localhost:5432/db"
+  provider = "postgresql"
+  url      = "postgresql://localhost:5432/db?sslmode=disable"
+  default  = true
 }
 
-model Customer {
-  age      pg.SmallInt
-  amount   pg.Money
-  name     pg.Varchar(n: 10)
-  location pg.Point(y: 5, x: 6)
+datasource ms {
+  provider = "mysql"
+  url      = "mysql://localhost:3306/db"
+}
+
+model User {
+  id        UUID   @id
+  firstName String
+  settings  JSONB
+}
+
+model Post {
+  id        Int             @id
+  title     ms.VarChar(255)
+  wordCount ms.SmallInt
 }
 ```
 
+`User` has implicit PostgreSQL scoping due to `default = true`. Post uses `ms.` in front of `ms.VarChar(255)` & `ms.SmallInt` to explicitly choose the MySQL datatypes.
```

Comment:
This needs to be rather specified on a model, instead of a field level. I'd suggest a syntax like
```
model Post {
  // ...
  @@datasource(ms)
}
```

Reply:
okay, i find this to be a bit goofy, it's like javascript's amusing `with` statement, but i don't have a strong preference either way. in general, we're chipping away at the original motivation for the schema which was a dependency graph, `ms.VarChar(255)` does a better job specifying that this type depends on this data source in my opinion.

Also if we want polyglot models, the proposed syntax would probably make more sense.